Make shield generator compatible with ES2019

The maplibre-shield-generator library is built for `ESNext` (which would be ECMAScript 2023 at the time of writing):

https://github.com/ZeLonewolf/openstreetmap-americana/blob/4ba85f903250b1747688d9729f0059d95dba0288/tsconfig.json#L3-L4

This build setting is so bleeding-edge that it could apparently open the door to [code that no released browser can parse at all](https://github.com/evanw/esbuild/releases/tag/v0.18.7) if we upgrade esbuild too soon. A more immediate issue is that even some of the recent ECMAScript versions would be incompatible with websites that might be interested in adopting the library.

For example, the OpenStreetMap Wiki recently integrated maplibre-shield-generator, loading it dynamically on [pages that embed OSM Americana](https://wiki.openstreetmap.org/wiki/Special:PermanentLink/2551555#New_templates_for_vector_maps). This makes it the first “core” OSM website to adopt client-side-rendered vector maps. Unfortunately, the mere presence of this library breaks all JavaScript for users of browsers such as Firefox 91 ESR (released in 2021). Since the library is compiled as a module, it becomes necessary to [dynamically import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) it from another script, which leads to the following error:

> SyntaxError: import declarations may only appear at top level of a module

The OSM Wiki currently runs [MediaWiki 1.39](https://wiki.openstreetmap.org/wiki/Special:Version), which [provides “grade-A” support](https://www.mediawiki.org/wiki/Special:PermanentLink/6012138#Browser_support_matrix) for “modern browsers”, defined as the last three years of Chromium releases, two LTS releases of Firefox, and Safari 11 (2018). Scripts are loaded by [ResourceLoader](https://www.mediawiki.org/wiki/ResourceLoader/Core_modules#mw.loader.load), a built-in bundler and minifier. ResourceLoader bundles maplibre-shield-generator with other site scripts, such as the one that powers taginfo statistics and taglists.

The OSM Wiki isn’t required to support every configuration that MediaWiki does, but Americana is currently embedded on so few pages that limiting browser support based on it would be quite a tail-wagging-the-dog situation. I can also imagine other websites besides wikis facing similar hurdles from how an ESNext-targeted library might interact with other dependencies.

It would make a lot of sense for maplibre-shield-generator, as a MapLibre GL JS plugin, to support the same browsers that MapLibre GL JS does. As far as I know, MapLibre GL JS lacks a clearly articulated browser compatibility policy, but its build system is configured to target [ES2019](https://github.com/maplibre/maplibre-gl-js/blob/30c0e1d47f019f159edc6c6ef8c5e068f14a696e/tsconfig.json#L17).

If I set `target` to `es2021` and `module` to `es2020`, running `npm run build:code` produces this error:

```
/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/index.ts:859
    return new TSError(diagnosticText, diagnosticCodes, diagnostics);
           ^
TSError: ⨯ Unable to compile TypeScript:
scripts/build.ts:86:1 - error TS1378: Top-level 'await' expressions are only allowed when the 'module' option is set to 'es2022', 'esnext', 'system', 'node16', or 'nodenext', and the 'target' option is set to 'es2017' or higher.

86 await build();
   ~~~~~

    at createTSError (/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/index.ts:859:12)
    at reportTSError (/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/index.ts:863:19)
    at getOutput (/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/index.ts:1077:36)
    at Object.compile (/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/index.ts:1433:41)
    at transformSource (/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/esm.ts:400:37)
    at /Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/esm.ts:278:53
    at async addShortCircuitFlag (/Users/mxn/Documents/Code/hub/openstreetmap-americana/node_modules/ts-node/src/esm.ts:409:15)
    at async nextLoad (node:internal/modules/esm/loader:161:22)
    at async ESMLoader.load (node:internal/modules/esm/loader:594:20)
    at async ESMLoader.moduleProvider (node:internal/modules/esm/loader:446:11) {
  diagnosticCodes: [ 1378 ]
}
```

Would it be possible to compile the shield library with more lenient options than the rest of the Americana package, since it needs wider distribution? Or would that be dependent on moving the shield library to a separate repository (#889)?

--- a/shieldlib/package.json
+++ b/shieldlib/package.json
@@ -1,7 +1,7 @@
 {
   "name": "@americana/maplibre-shield-generator",
   "description": "Generate highway shields for maplibre-gl-js maps",
-  "version": "0.0.2",
+  "version": "0.0.3",
   "author": "OpenStreetMap Americana Contributors",
   "type": "module",
   "keywords": [

--- a/shieldlib/tsconfig.json
+++ b/shieldlib/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
-    "target": "ESNext",
-    "module": "ESNext",
+    "target": "ES2019",
+    "module": "ES2020",
     "moduleResolution": "node",
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true

